spring-core/src/main/java/org/springframework/aot/hint/nativex/NativeConfigurationGenerator.java

```diff
@@ -0,0 +1,160 @@
+/*
+ * Copyright 2002-2022 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.aot.hint.nativex;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Path;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.SerializationFeature;
+import com.fasterxml.jackson.databind.module.SimpleModule;
+
+import org.springframework.aot.hint.JavaSerializationHints;
+import org.springframework.aot.hint.ProxyHints;
+import org.springframework.aot.hint.ReflectionHints;
+import org.springframework.aot.hint.ResourceHints;
+import org.springframework.aot.hint.RuntimeHints;
+
+/**
+ * Generate the GraalVM native configuration files from runtime hints.
+ *
+ * @author Sebastien Deleuze
+ * @since 6.0
+ * @see <a href="https://www.graalvm.org/22.0/reference-manual/native-image/BuildConfiguration/">Native Image Build Configuration</a>
+ */
+public class NativeConfigurationGenerator {
+
+	private final ObjectMapper objectMapper;
+
+
+	public NativeConfigurationGenerator() {
+		this(new ObjectMapper().enable(SerializationFeature.INDENT_OUTPUT));
+	}
+
+	/**
+	 * Constructor with a Jackson {@link ObjectMapper} to use instead of using the default.
+	 */
+	public NativeConfigurationGenerator(ObjectMapper objectMapper) {
+		this.objectMapper = objectMapper;
+		registerSerializers();
+	}
+
+	private void registerSerializers() {
+		SimpleModule module = new SimpleModule();
+		module.addSerializer(JavaSerializationHints.class, new JavaSerializationHintsSerializer());
+		module.addSerializer(ProxyHints.class, new ProxyHintsSerializer());
+		module.addSerializer(ReflectionHints.class, new ReflectionHintsSerializer());
+		module.addSerializer(ResourceHints.class, new ResourceHintsSerializer());
+		this.objectMapper.registerModule(module);
+	}
+
+	/**
+	 * Generate the native configuration files for the provided hints.
+	 * @param hints the hints to serialize
+	 * @param targetDirectory directory where the {@code /META-INF/native-image/*-config.json} files should be generated
+	 */
+	public void generate(RuntimeHints hints, Path targetDirectory) {
```

Comment:
I wonder if a strategy interface here would be more appropriate. The generator is focusing on file system but that should be an implementation detail. It would be nice also if the [notion of namespacing](https://www.graalvm.org/22.0/reference-manual/native-image/BuildConfiguration/#embedding-a-configuration-file) was a bit more evident. 

For instance, an implementation could take a groupId, an artifactId and then generate the `META-INF/native-image/{groupId}/{artifactId}` structure based on a base path.